src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,120 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_INDEX = 10;
+    private static final double GROW_INDEX = 1.5;
+    private int maxSize = DEFAULT_INDEX;
+    private int sizeOfArray;
+    private T[] arrayDefault;
+    private T[] arrayTemporary;
+
+    public ArrayList() {
+        arrayDefault = (T[]) new Object[maxSize];
+    }
+
+    public void errorCheck(int index, int size) throws ArrayListIndexOutOfBoundsException {
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("No such index exception: "
+                    + index + " for size" + size);
+        }
+    }
 
+    public void resize() {
+        if (sizeOfArray == maxSize) {
+            maxSize = (int) (maxSize * GROW_INDEX);
+            arrayTemporary = (T[]) new Object[maxSize];
+            System.arraycopy(arrayDefault, 0, arrayTemporary, 0, sizeOfArray);
+            arrayDefault = arrayTemporary;
+        }
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T value) {
+        resize();
+        for (int i = 0; i < sizeOfArray; i++) {
+            if ((arrayDefault[i] != null && arrayDefault[i] == value)
+                    || (arrayDefault[i] != null && arrayDefault[i].equals(value))) {
+                arrayDefault[i] = value;
+                return;
+            }
+        }
+        arrayDefault[sizeOfArray] = value;
+        sizeOfArray++;
+    }
 
+    @Override
+    public void add(T value, int index) throws ArrayListIndexOutOfBoundsException {
```

Comment:
```suggestion
    public void add(T value, int index) {
```
it is a runtime exception no need to put it in method signature

Reply:
not fixed